### PR TITLE
[fix] [io] elastic-search sink connector not support JSON.String schema.

### PR DESCRIPTION
**Pull request by shibd**
### Motivation

If a producer use `Schema.JSON(String.class)` schema, es sink connector will throw `ClassCastException`.

1. Create a producer.
```java
Producer<String> producer = client.newProducer(Schema.JSON(String.class))
        .topic("test-es")
        .create();
MessageId msgID = producer.send({\"a\":1});
```

2. When es sink connector handle this message, it will throw `ClassCastException`.
```
java.lang.ClassCastException: class java.lang.String cannot be cast to class com.fasterxml.jackson.databind.JsonNode (java.lang.String is in module java.base of loader 'bootstrap'; com.fasterxml.jackson.databind.JsonNode is in unnamed module of loader 'app')
```
Exception throws on here:
https://github.com/apache/pulsar/blob/593fcb87b6d3bd6401eb4ffa4c25d8b3f724e49d/pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java#L382

The root cause is when the schema is not `RECORD` type, it will not transfer to `JsonNode`.
https://github.com/apache/pulsar/blob/af1360fb167c1f9484fda5771df3ea9b21d1440b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java#L204-L220

### Modifications

- When `navtiveObject` is String, try to transfer it to JsonNode.

### Verifying this change

- Add `sendJsonStringSchemaTest ` to cover it.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/shibd/pulsar/pull/22

### CODE REVIEW

**Review by poorbarcode — APPROVED** [review]
LGTM

**Comment by poorbarcode** [comment]
@shibd 

Could you add the labels which branches should cherry-pick into?

File: pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSinkTests.java [hunk 1]
```diff
@@ -243,6 +243,46 @@ public Schema answer(InvocationOnMock invocation) throws Throwable {
...
+            public Schema answer(InvocationOnMock invocation) throws Throwable {
+                return Schema.JSON(String.class);
+            }
+        });
+
+        map.put("indexName", "test-index");
+        map.put("schemaEnable", "true");
+        sink.open(map, mockSinkContext);
+        sink.write(mockRecord);
+        verify(mockRecord, times(1)).ack();
```
Comment (congbobo184) [inline pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSinkTests.java:283]: may be we should add a test for the exception that you added

Comment (shibd) [reply pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSinkTests.java:284]: Thanks, added a test for converting JSON exception, PTAL.

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**